apps/web/src/core/providers/web-vm.ts

```diff
@@ -0,0 +1,388 @@
+import { createProvider } from "puro"
+import { useContext, useEffect, useRef, useState } from "react"
+import { type Terminal } from "xterm"
+
+import { WebContainer, type WebContainerProcess } from "@webcontainer/api"
```

Comment:
I should come back to re-review this once i can test the preview pane, which isn't working for me